lib/bots/versionbot.ts

```diff
@@ -825,12 +861,23 @@ export class VersionBot extends GithubBot.GithubBot {
                 // Get the list of commits for the PR, then get the very last commit SHA.
                 return this.versionBotCommits(prInfo).then((commitMessage: string | null) => {
                     if (commitMessage) {
-                        // We go ahead and merge.
-                        return this.mergeToMaster({
-                            commitVersion: commitMessage,
-                            owner,
-                            prNumber: prInfo.number,
-                            repoName: repo
+                        // Ensure that the labeler was authorised. We do this here, else we could
+                        // end up spamming the PR with errors.
+                        return this.getConfiguration(owner, repo).then((config: VersionBotConfiguration) => {
+                            // If this was a labeling action and there's a config, check to see if there's a maintainers
+                            // list and ensure the labeler was on it.
+                            // This throws an error if not.
+                            if ((data.action === 'labeled') && config) {
+                                this.validMaintainer(config, data);
```

Comment:
You don't need the `&& config` check, both here and before.

`validMaintainer` already handles the case where there's no config or the config doesn't have maintainers. Its arg types don't show it right now, but if you just make VersionBotConfig optional you can drop this.